plugin/pkg/scheduler/factory/factory.go

```diff
@@ -257,16 +257,17 @@ func getNodeConditionPredicate() cache.NodeConditionPredicate {
 
 // Returns a cache.ListWatch that finds all pods that need to be
 // scheduled.
-func (factory *ConfigFactory) createUnassignedPodLW() *cache.ListWatch {
-	return cache.NewListWatchFromClient(factory.Client, "pods", api.NamespaceAll, fields.Set{client.PodHost: ""}.AsSelector())
+func (factory *ConfigFactory) createUnassignedAndRunningPodLW() *cache.ListWatch {
```

Comment:
Can we rename it to "createNonTerminatedPodLW"?
In fact it's mostly what you are checking (phase != succeeded and phase != failed)